src/corrgi/correlation/projected_correlation.py

```diff
@@ -1,22 +1,55 @@
 from typing import Callable
 
+import gundam.cflibfor as cff
+import numpy as np
 import pandas as pd
+from astropy.cosmology import LambdaCDM
+from gundam import gundam
 from hipscat.catalog.catalog_info import CatalogInfo
+from munch import Munch
 
 from corrgi.correlation.correlation import Correlation
 
 
 class ProjectedCorrelation(Correlation):
     """The projected correlation utilities."""
 
+    def __init__(self, params: Munch, use_weights: bool = False):
+        super().__init__(params, use_weights)
+        self.cosmo = LambdaCDM(H0=params.h0, Om0=params.omegam, Ode0=params.omegal)
+        self.sepp, self.sepv = self.make_bins()
+
+    def make_bins(self) -> tuple[list]:
+        """Generate bins of projected separation and LOS for the correlation"""
+        sepp, _ = gundam.makebins(
+            self.params.nsepp, self.params.seppmin, self.params.dsepp, self.params.logsepp
+        )
+        sepv, _ = gundam.makebins(self.params.nsepv, 0.0, self.params.dsepv, False)
+        return sepp, sepv
+
+    def calculate_comoving_distances(self, df: pd.DataFrame) -> np.ndarray:
+        """Calculate the comoving distances from the redshift of each particle"""
+        return self.cosmo.comoving_distance(df["z"].to_numpy()).value
```

Comment:
What if there's no "z" column?

Reply:
I added a validation step that can be overridden by each correlation function.